### PR TITLE
fix(tokens): keep usage summary counts and costs consistent

### PR DESCRIPTION
**Pull request by yashb98**
I found that a model call finishing during a pricing lookup can make `get_usage_summary()` disagree with itself. The total token count covers the earlier calls, while the costs and per-model totals include the new call.

I copy the usage history before calculating the summary so every field uses the same set of calls. Calls added during the lookup appear in the next summary. The tests control this timing and cover summaries with and without filters.

## Validation

On Python 3.12.11, the regression tests gave 1 failure and 3 passes before the fix. With the fix, all 14 focused tests and the applicable pre-commit checks passed. Pricing is mocked; no live model or browser was used. I did not run the full repository suite.

```bash
uv run --no-sync pytest \
  tests/ci/test_token_usage_summary.py \
  tests/ci/test_openrouter_token_cost.py \
  tests/ci/test_orcarouter.py -o addopts= -q --tb=short
uv run --no-sync pre-commit run --files \
  browser_use/tokens/service.py tests/ci/test_token_usage_summary.py
```

---
## Summary by cubic
Fixes a race condition where a model call finishing mid-pricing made usage summaries disagree with themselves, and keeps logged summaries consistent with what was shown. Summaries and logs now operate on one fixed snapshot of usage history, with calls that complete during the lookup deferred to the next summary.

`log_usage_summary()` now uses the summary's computed per-model cost breakdowns instead of re-pricing history entries. `ModelUsageStats` gains `prompt_cost` and `completion_cost` fields (`None` when pricing is unavailable, `0.0` when pricing is free).

**Migration**
- No breaking changes; summaries may differ from before only by excluding calls that completed mid-lookup.

<sup>Written for commit a4025956d82662c47af48a403fa69c1fef1a65ac. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/browser-use/browser-use/pull/5701?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: browser_use/tokens/service.py [hunk 1]
```diff
@@ -425,7 +425,8 @@ def get_usage_tokens_for_model(self, model: str) -> ModelUsageTokens:
 
 	async def get_usage_summary(self, model: str | None = None, since: datetime | None = None) -> UsageSummary:
 		"""Get summary of token usage and costs (costs calculated on-the-fly)"""
-		filtered_usage = self.usage_history
+		# Keep token totals and costs on the same snapshot while pricing is awaited.
+		filtered_usage = list(self.usage_history)
```
[the remark this answers is not in the document]
Comment (yashb98) [reply browser_use/tokens/service.py:429]: Fixed in f780cc7c72adf182a781d1f5056019a7f8cdd8b0. I changed `log_usage_summary()` to use the costs in `summary.by_model`, so it no longer prices usage a second time. Optional per-model input/output costs preserve the breakdown, including in the beta producer. Summaries without these fields still load.

Locally, 100 selected repository tests and seven additional edge-case checks passed. Eight timing cases also passed in ten separate runs. Applicable pre-commit checks passed, including lint and type checks. I did not run the full suite or live-provider tests.
